### PR TITLE
Support for UWP

### PR DESCRIPTION
**Pull request by TomSoderling**
This is my attempt at support for Universal Windows Platform (UWP).  I followed a pretty basic monkey-see, monkey-do approach and added a new NGraphics.UWP project that relies on the existing implementation for the WinRT platform which seems to work well on UWP also.
I included a new test project for UWP and it produces test files as I'd expect
Took this opportunity to update to the brand spankin' new SharpDX v3.1.0 library as well
I believe VS 2015 update 3 is required to open and run the UWP project

Let me know if I screwed something up and I'll try to fix it

### CODE REVIEW

File: NGraphics.Test/GraphicCanvasTests.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public void Ellipse ()
 		{
 			var d = new GraphicCanvas (new Size (50, 50), Platform);
 			d.DrawEllipse (new Point (10, 20), new Size (30, 40), Pens.Black);
-			Assert.AreEqual (1, d.Graphic.Children.Count);
+			Assert.AreEqual (1f, d.Graphic.Children.Count);
```
Comment (laarmen) [inline NGraphics.Test/GraphicCanvasTests.cs]: Err, why in heaven would you want to compare a Count property to a float?

Comment (TomSoderling) [inline NGraphics.Test/GraphicCanvasTests.cs]: Weird, right?

**Comment by laarmen** [comment]
Unless I am mistaken it is not possible to test this PR! At least, I can't see the repo/branch it originates from, which means I can't pull the code.

**Comment by TomSoderling** [comment]
Ah, thanks for pointing that out. I didn't realize it was saying "from unknown repository".  I wonder if that's because I'm part of an organization github account?
This is the source branch, under my company Mobile Composer: https://github.com/MobileComposer/NGraphics

**Comment by squinn7** [comment]
Is there any update on this PR? I would be very interested in seeing UWP support but it doesn't seem like this request is being updated anymore.

**Comment by praeclarum** [comment]
Sorry for the long delays and thank you for the great work @TomSoderling!

**Comment by TomSoderling** [comment]
Fantastic! Thank you!

**Comment by jasells** [comment]
Is there an ETA on the next release?  I reeealy need UWP support.

**Comment by TomSoderling** [comment]
I'd try reaching out to Frank on twitter: @praeclarum

**Comment by gabornemeth** [comment]
I feel 0.5.0-beta1 works well on UWP too. :thumbsup:
I managed to update [my fork of NControl](https://github.com/gabornemeth/NControl/tree/uwp) with it.

I noticed ```SurfaceImageSourceCanvas``` in ```NGraphics.UWP``` and tried to use it. It does not seem to work right now. Is it planned to be used? I played around with [it](https://gist.github.com/gabornemeth/4b9b34d7c352e86c65f2e0cf30616f62) trying to use [this way](https://gist.github.com/gabornemeth/0c3f0a44bde69a846f5073ab90e88961). It showed some signs of operation. Finally I gave up that and fell back using the ```ICanvas``` returned by the Platform.

**Comment by jasells** [comment]
Just FYI, I have abandoned Ncontrol, because of the apparent lack of activity, and what I believe to be a more viable alternative.

[SkiaSharp](https://blog.xamarin.com/cross-platform-2d-graphics-with-skiasharp/)

[More info](https://developer.xamarin.com/guides/cross-platform/drawing/)

No slam on NControl, but my assumption for the lack of activity in general has been that many people are making the same decision.  Skia is actively maintained by google, is a cross-platform API (as opposed to NGraphics using 'native' API's for each platform) so it will be easier to maintain.  SkiaSharp is maintained by the Xamarin team (Microsoft), so I feel much more confident it will continue to be viable.

My team is almost finished porting our NControl code to SkiaSharp, and it has not been too difficult.

Another advantage of SkiaSharp is that it is also compatible with WPF and Windows forms, if you want to develop Linux/Windows desktop apps.

 gabornemeth, Thanks for your efforts, at the very least it is a possible short term solution for those who can't or don't want to move from NControl yet.

**Pull request merged**